### DISCUSSION TITLE
Support for Macbook trackpad "Force Touch"

### DISCUSSION THREAD

**Discussion by SproedKartoffelChip** [YaLTeR/niri#729] [Ideas]
On MacOS I use BetterTouchTool to bind unused trackpad inputs to many different actions, including a three finger "Force Touch" to close a window and four fingers to quit an application. Being able to utilize the pressure sensitivity on my trackpad for alternate actions on niri would be really nice, especially once gesture binding is implemented as it would help me exclusively manage my windows with my hands off the keyboard. 

As far as I can tell using `libinput measure touchpad-pressure` on Asahi Fedora Remix the drivers do support pressure sensitivity and send a different input labelled "down" when applying pressure that is well above the regular clicking threshold, but there isn't any haptic feedback once you reach the threshold for the "down" input.

**Comment by YaLTeR** [comment]
I'm not sure how libinput reports these events. Can you bind them in any other compositor?

**Reply by SproedKartoffelChip** [reply]
No and it doesn't seem like an equivalent implementation of "Force Touch" at all exists anywhere in Linux from 2 or so hours of research. Even if you manage to support binding an input to a certain touch pressure threshold you'd still have no way to give any kind of haptic feedback. AFAIK with the current drivers it's not even possible to do anything with the haptic motors like MacOS lets you do: https://developer.apple.com/documentation/appkit/nshapticfeedbackmanager 

oh well :/

**Reply by YaLTeR** [reply]
Right, that's unfortunate..
